docker-compose.yml

```diff
@@ -955,6 +956,20 @@ services:
         /arrow/ci/scripts/python_build.sh /arrow /build &&
         /arrow/ci/scripts/python_test.sh /arrow"
 
+  ubuntu-d:
+    # Usage:
+    #   docker-compose build ubuntu-d
+    #   docker-compose run --rm ubuntu-d
+    # Parameters:
+    image: ubuntu-d
+    build:
+      context: .
+      dockerfile: ci/docker/ubuntu-d.dockerfile
+    shm_size: *shm-size
+    volumes: *ubuntu-volumes
+    command: >
+      /bin/bash -c "/arrow/ci/scripts/d_test.sh /arrow"
```

Comment:
Could you build Apache Arrow C++ and GLib before `d_test.sh`?
We can use `ci/scripts/{cpp,c_glib}_build.sh` like other services.